ckanext/qdes_schema/validators.py

```diff
@@ -305,6 +309,12 @@ def qdes_iso_8601_durations(key, flattened_data, errors, context):
     if has_error:
         raise toolkit.Invalid('The value in each field needs to be positive number.')
 
+    # Validate the pattern.
+    result_pattern = re.split("^P(?=\d+[YMWD])(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(?=\d+[HMS])(\d+H)?(\d+M)?(\d+S)?)?$", flattened_data[key])
```

Comment:
Fails validation on the UI if no value is set on the form

Reply:
refactored, added `ignore_empty ignore_missing`